### ISSUE TITLE
Render-theme with "mtb:scale:uphill" and "mtb:scale" displays wrong tags

### ISSUE THREAD

**Issue by mapsforge** [mapsforge/mapsforge#353]
_From [frank.hu...@gmail.com](https://code.google.com/u/109115453950959521277/) on September 18, 2012 21:58:02_

What steps will reproduce the problem? 1. Create a small xml (osm) file rttesttrack.xml, modify some tracks and add mtb:scale and mtb:scale:uphill tags. Tags should be different (e.g mtb:scale = 1, mtb:scale:uphill=2)
1. Add the following lines after line 386 to the original tag-mapping.xml and save it as mtbtag-mapping.xml:
   
   ```
   <osm-tag key="mtb:scale" value="0" renderable="true" />
   <osm-tag key="mtb:scale" value="1" renderable="true" />
   <osm-tag key="mtb:scale" value="2" renderable="true" />
   <osm-tag key="mtb:scale" value="3" renderable="true" />
   <osm-tag key="mtb:scale" value="4" renderable="true" />
   
   <osm-tag key="mtb:scale:uphill" value="0" renderable="true" />
   <osm-tag key="mtb:scale:uphill" value="1" renderable="true" />
   <osm-tag key="mtb:scale:uphill" value="2" renderable="true" />
   <osm-tag key="mtb:scale:uphill" value="3" renderable="true" />
   <osm-tag key="mtb:scale:uphill" value="4" renderable="true" />
   ```
2. Convert the modified osm file with osmosis/mapwriter: 
   osmosis-0.40.1/bin/osmosis --rx file=rttesttrack.xml --mw file=rttesttrack.map
   bbox=48.48,9.17,48.5,9.19 tag-conf-file=mtbtag-mapping.xml
3. Modify an existing render theme (I took the bluenight-theme from Locus forum), add the lines to show up symbols for MTB (for those interested, I tried to make the same symbols as in openmtbmap.org map for Garmin devices). I will not show all tags, but the idea is: 

<rule e="way" k="highway" v="track|path">
  <rule e="way" k="mtb:scale" v="0" zoom-min="16">
    <lineSymbol src="file:/symbols/mtbscale0.png"
        align-center="false" repeat="true" />
  </rule>
  <rule e="way" k="mtb:scale" v="1" zoom-min="16">
    <lineSymbol src="file:/symbols/mtbscale1.png"
        align-center="false" repeat="true" />
  </rule>

  <rule e="way" k="mtb:scale:uphill" v="0" zoom-min="16">
    <lineSymbol src="file:/symbols/mtbscaleuphill0.png"
        align-center="false" repeat="true" />
  </rule>
</rule>

(I added the same rules for all scales (0 to 4) with the respective symbol)
1. Opened Locus, used the new map file to show up the map with with the new tags shown as expected What is the expected output? What do you see instead? I do not see the correct tags. It looks like  mtb:scale:uphill value overided the mtb:scale value: A track with mtb:scale set to 0 and mtb:scale:uphill set to 4 shows up with symbols for mtb:scale=4 and mtb:scale:uphill=4

The map shows up (also with other render themes), no errors. What version of the product are you using? On what operating system? I do use the current map-writer (0.30), osmosis 0.40.1 and Locus Pro 2.6.1 Please provide any additional information below. So far I did not check the MapViewer.apk. I will check this later and add my findings.

_Original issue: http://code.google.com/p/mapsforge/issues/detail?id=353_

**Comment by devemux86** [comment]
Mapsforge has changed a lot since then, can create new issue if still valid.

**Comment by devemux86** [comment]
Fixed in #1366.
